user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/BatchDetailsComponent.java

```diff
@@ -0,0 +1,375 @@
+  }
+
+  private void updateBatch(BatchPreview batchPreview, boolean fromClient) {
+    EditBatchEvent editBatchEvent = new EditBatchEvent(this, batchPreview, fromClient);
+    fireEvent(editBatchEvent);
+  }
+
+  private void removeBatch(BatchPreview batchPreview, boolean fromClient) {
+    DeleteBatchEvent deleteBatchEvent = new DeleteBatchEvent(this, batchPreview, fromClient);
+    fireEvent(deleteBatchEvent);
+  }
+
+
+  private void loadBatchesForExperiment(Experiment experiment) {
+    batchInformationService.retrieveBatchesForExperiment(experiment.experimentId())
+        .onValue(batches -> batchPreviews.addAll(
+            batches.stream().map(this::generatePreviewFromBatch).toList()));
+  }
+
+  /**
+   * Register a {@link ComponentEventListener} that will get informed with an
+   * {@link ViewBatchEvent}, as soon as a user wants to view a {@link Batch}
+   *
+   * @param batchViewListener a listener on the batch view trigger
+   */
+  public void addBatchViewListener(
+      ComponentEventListener<ViewBatchEvent> batchViewListener) {
+    addListener(ViewBatchEvent.class, batchViewListener);
+  }
+
+  /**
+   * Register a {@link ComponentEventListener} that will get informed with an
+   * {@link CreateBatchEvent}, as soon as a user wants to create a {@link Batch}
+   *
+   * @param batchCreationListener a listener on the batch creation trigger
+   */
+  public void addBatchCreationListener(
+      ComponentEventListener<CreateBatchEvent> batchCreationListener) {
+    addListener(CreateBatchEvent.class, batchCreationListener);
+  }
+
+  /**
+   * Register an {@link ComponentEventListener} that will get informed with a
+   * {@link EditBatchEvent}, as soon as a user wants to edit batch Information.
+   *
+   * @param batchEditListener a listener on the batch edit trigger
+   */
+  public void addBatchEditListener(ComponentEventListener<EditBatchEvent> batchEditListener) {
+    addListener(EditBatchEvent.class, batchEditListener);
+  }
+
+  /**
+   * Register a {@link ComponentEventListener} that will get informed with an
+   * {@link DeleteBatchEvent}, as soon as a user wants to delete a {@link Batch}
+   *
+   * @param batchDeletionListener a listener on the batch deletion trigger
+   */
+  public void addBatchDeletionListener(
+      ComponentEventListener<DeleteBatchEvent> batchDeletionListener) {
+    addListener(DeleteBatchEvent.class, batchDeletionListener);
+  }
+
+  private LocalDateTime asClientLocalDateTime(Instant instant) {
+    ZonedDateTime zonedDateTime = instant.atZone(ZoneId.of(
+        this.clientDetailsProvider.latestDetails().map(ClientDetails::timeZoneId).orElse("UTC")));
+    return zonedDateTime.toLocalDateTime();
+  }
+
+  public static class BatchPreview implements Serializable {
+
+    @Serial
+    private static final long serialVersionUID = 5781276711398861714L;
+    private BatchId batchId;
+    private String batchLabel;
+    private Instant creationDate;
+    private Instant modificationDate;
```

Comment:
As we do not have the timepoints anywhere, can we remove it for now?

Reply:
Done